### PR TITLE
DRY shader creation via helper lambda

### PR DESCRIPTION
**Pull request by vittorioromeo**
Should not change any existing behavior, just DRY refactoring. Useful to clearly demonstrate that there are no difference between the creation/attachment of different shader types.

### CODE REVIEW

File: src/SFML/Graphics/Shader.cpp [hunk 1]
```diff
@@ -868,89 +868,49 @@ bool Shader::compile(std::string_view vertexShaderCode, std::string_view geometr
     // Create the program
     const GLEXT_GLhandle shaderProgram = glCheck(GLEXT_glCreateProgramObject());
 
-    // Create the vertex shader if needed
-    if (!vertexShaderCode.empty())
+    // Helper function for shader creation
+    const auto createAndAttachShader = [&](GLenum shaderType, const char* shaderTypeStr, std::string_view shaderCode)
```
Comment (ChrisThrasher) [inline src/SFML/Graphics/Shader.cpp:872]:
```suggestion
    const auto createAndAttachShader = [shaderProgram](GLenum shaderType, const char* shaderTypeStr, std::string_view shaderCode)
```
If a lambda only captures 1 or 2 things I'd prefer we explicitly capture those variables.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]
One nitpick but LGTM

File: src/SFML/Graphics/Shader.cpp (hunk 1 above)
Comment (vittorioromeo) [reply src/SFML/Graphics/Shader.cpp:872]: Good point :)

**Review by ChrisThrasher — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Pull request merged**